addon/components/mdl-textfield.js

```diff
@@ -12,5 +18,15 @@ export default BaseComponent.extend({
   ],
   beforeMdlInit() {
     this.$('label.mdl-button').attr('for', this.get('_inputId'));
-  }
+  },
+  setValidity: observer('errorMessage', function() {
+    let mdlComponent = this.get('_mdlComponent');
+    let errorMessage = this.get('errorMessage');
+
+    if (isPresent(errorMessage)) {
+      mdlComponent.input_.setCustomValidity(errorMessage.toString());
+    } else {
+      mdlComponent.input_.setCustomValidity('');
+    }
```

Comment:
@stefanpenner - Interested in your opinion of proper observer usage (I don't think we have any bad observers at all in this addon yet -- would like to keep it that way). 

Should this be debounced or something, in order to ensure that we don't end up with the typical "sloshing" problem?